### DISCUSSION TITLE
No id generated for recordings

### DISCUSSION THREAD

**Discussion by kkrystev** [opendatacam/opendatacam#561] [Q&A]
The software is up and running on Ubuntu 22.04 with nvidia 1050 using doker setup.
The problem is that when I start a recording, the counters work normal and count correct, but I cannot export anything from data tab: csv or json.
Before running the recording /status returns:
<HTML><BODY><div class="toolbar"><div class="devtools-separator"></div><div class="devtools-searchbox"><input class="searchBox devtools-filterinput" placeholder="Filter JSON" value=""></div></div><div class="panelContent">

  |  
-- | --
counterSummary | {}
trackerSummary |  
totalItemsTracked | 49
videoResolution |  
w | 640
h | 360
appState |  
yoloStatus |  
isStarting | false
isStarted | true
isListeningToYOLO | true
recordingStatus |  
requestedFileRecording | false
isRecording | false
currentFPS | 17
recordingId | null
dateStarted | null
filename | ""

</div></BODY></HTML>

and after start of the recording it returns:

<HTML><BODY><li class="tabs-menu-item json is-active" role="presentation"><span class="devtools-tab-line"></span></li><li class="tabs-menu-item rawdata " role="presentation"><span class="devtools-tab-line"></span></li><span class="devtools-tab-line"></span><div id="json-panel" style="visibility: visible; height: 100%;" class="tab-panel-box" role="tabpanel" aria-labelledby="json-tab"><div class="tab-panel json"><div class="jsonPanelBox tab-panel-inner"><div class="toolbar"></div></div></div></div><div class="toolbar"><div class="devtools-separator"></div><div class="devtools-searchbox"><input class="searchBox devtools-filterinput" placeholder="Filter JSON" value=""></div></div><div class="panelContent">

  |  
-- | --
counterSummary |  
45e39581-243c-45cc-99e5-3c790931e10c |  
_total | 9
car | 9
trackerSummary |  
totalItemsTracked | 392
videoResolution |  
w | 640
h | 360
appState |  
yoloStatus |  
isStarting | false
isStarted | true
isListeningToYOLO | true
recordingStatus |  
requestedFileRecording | false
isRecording | true
currentFPS | 17
recordingId | null
dateStarted | "2022-09-09T14:49:18.821Z"
filename | "demo.mp4"

</div></BODY></HTML>

After stopping the recording the result in the console is:
(node:40) UnhandledPromiseRejectionWarning: MongoServerSelectionError: getaddrinfo EAI_AGAIN mongo
    at Timeout._onTimeout (/var/local/opendatacam/node_modules/mongodb/lib/core/sdam/topology.js:430:30)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)
(node:40) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 5330)

The issue in my opinion is that there is no ID for the recording.
Could you please advice me something?
Thank you in advance.

**Comment by vsaw** [comment]
The error message you posted suggests that either, MongoDB is not running, or you are pointing OpenDataCam in the wrong direction.

The default value for MongDB in the config.json are for use with docker compose. If you do not use docker compose you most likely have MongoDB running locally, in this case you should change the hostname from `mongo` to `localhost`.

**Reply by kkrystev** [reply]
I also guess that the issue is coming from MongoDB. It's a default installation, using docker compose.
I've tried to change mongo with localhost, just for the check and here is the error message:

(node:40) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 133)
(node:40) UnhandledPromiseRejectionWarning: MongoServerSelectionError: connect ECONNREFUSED 127.0.0.1:27017
    at Timeout._onTimeout (/var/local/opendatacam/node_modules/mongodb/lib/core/sdam/topology.js:430:30)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)

What do you mean with pointing datacam in wrong direction?

**Reply by vsaw** [reply]
If you are using docker-compose, then 

> "MONGODB_URL": "mongodb://mongo:27017"

should be correct. Chances are MongoDB did not start up correctly. Did you check the logs of the MongoDB container? Maybe this gives you a hint.
